woocommerce/wc-admin
# Activity Panel: Inbox: Add Woo Extension Subscription / Connection Notes

See #176 

Adds woo extension subscription and connection notes to the merchant inbox. Note: Bumping expiring notices at 45/20/7/1 days out will be done under a separate PR (#711) since this one got big.

READY FOR REVIEW / TESTING

### Screenshots

<img width="851" alt="screen shot 2018-10-25 at 3 09 22 pm" src="https://user-images.githubusercontent.com/1595739/47533270-fc3eb800-d867-11e8-8c68-bfe63c1abbaf.png">

<img width="717" alt="screen shot 2018-10-25 at 4 43 21 pm" src="https://user-images.githubusercontent.com/1595739/47536492-26e33d80-d875-11e8-9240-de43dd1c76bd.png">

<img width="707" alt="screen shot 2018-10-25 at 4 50 19 pm" src="https://user-images.githubusercontent.com/1595739/47536631-17b0bf80-d876-11e8-98c4-5f86f7f02d6f.png">

### Detailed test instructions:

*Connection Test*

 - You will need to be able to connect your site to WooCommerce.com for this test.
- Start with your site not connected to WooCommerce.com
- Go to wp-admin > WooCommerce > Dashboard > Inbox and make sure you have a *Connect to WooCommerce.com* note in your inbox.
- Connect to a WooCommerce.com
- Verify the *Connect to WooCommerce.com* note is no longer in your inbox.

*Expiring Test*

- At WooCommerce.com, buy a single site subscription for Product Add-Ons
- DO NOT ENABLE AUTORENEW FOR PRODUCT ADD ONS
- Modify the order on woocommerce.com so that Product Add-Ons expires about 30 days from today.
- Download, install, and activate Product-Add Ons
- Set the Product-Add Ons subscription to active on wp-admin > WooCommerce > Extensions > WooCommerce.com Subscriptions
- Go to wp-admin > WooCommerce > Dashboard > Inbox and make sure you have a note that the Product Add Ons subscription is expiring in 30 days.
- Set the Product-Add Ons subscription to inactive on wp-admin > WooCommerce > Extensions > WooCommerce.com Subscriptions
- Go to wp-admin > WooCommerce > Dashboard > Inbox and make sure you DO NOT have a note about the Product Add Ons subscription.

*Expired Test*
- IMPORTANT: Set the Product-Add Ons subscription to active on wp-admin > WooCommerce > Extensions > WooCommerce.com Subscriptions (or you will not be able to do so once it has expired)
- Modify the order on woocommerce.com so that Product Add-Ons expired last week.
- Go to wp-admin > WooCommerce > Dashboard > Inbox and make sure you have a note that the Product Add Ons subscription expired.
- Set the Product-Add Ons subscription to inactive on wp-admin > WooCommerce > Extensions > WooCommerce.com Subscriptions
- Go to wp-admin > WooCommerce > Dashboard > Inbox and make sure you DO NOT have a note about the Product Add Ons subscription.
